src/components/core/DynamicImage/DynamicImage.tsx

```diff
@@ -114,61 +141,77 @@ const DynamicImage = ({
   useEffect(() => {
     // Skip animation for subsequent loads where the image hasn't changed
     const previousImage = getPrevImage()
-    if (previousImage !== null && isImageEqual(previousImage, image)) {
+    if (previousImage !== null && isImageEqual(previousImage, source)) {
       return
     }
 
-    setPrevImage(image ?? null)
+    setPrevImage(source ?? null)
 
     if (isFirstImageActive) {
       setIsFirstImageActive(false)
-      setFirstImage(image)
+      setFirstImage(source)
       animateTo(firstOpacity, 1)
       animateTo(secondOpacity, 0)
     } else {
       setIsFirstImageActive(true)
-      setSecondImage(image)
+      setSecondImage(source)
       animateTo(firstOpacity, 0)
       animateTo(secondOpacity, 1)
     }
   }, [
     animateTo,
     firstOpacity,
     getPrevImage,
-    image,
+    source,
     isFirstImageActive,
     secondOpacity,
     setIsFirstImageActive,
     setPrevImage
   ])
 
+  const handleSetFirstSize = useCallback((event: LayoutChangeEvent) => {
+    setFirstSize(event.nativeEvent.layout.width)
+  }, [])
+
+  const handleSetSecondSize = useCallback((event: LayoutChangeEvent) => {
+    setSecondSize(event.nativeEvent.layout.width)
+  }, [])
+
   return (
-    <View>
+    <View style={[style, stylesProp?.root]}>
       <Animated.View
-        style={[styles.image, { opacity: firstOpacity }]}
-        onLayout={e => setFirstSize(e.nativeEvent.layout.width)}
+        style={[
+          stylesProp?.imageContainer,
+          styles.imageContainer,
+          { opacity: firstOpacity }
+        ]}
+        onLayout={handleSetFirstSize}
       >
         <ImageWithPlaceholder
-          image={firstImage}
-          style={[{ width: firstSize, height: firstSize }, style]}
+          source={firstImage}
+          style={[{ width: firstSize, height: firstSize }, stylesProp?.image]}
           usePlaceholder={usePlaceholder}
         />
       </Animated.View>
       <Animated.View
         style={[
-          styles.image,
+          stylesProp?.imageContainer,
+          styles.imageContainer,
           { opacity: secondOpacity, zIndex: isFirstImageActive ? -1 : 0 }
         ]}
-        onLayout={e => setSecondSize(e.nativeEvent.layout.width)}
+        onLayout={handleSetSecondSize}
       >
         <ImageWithPlaceholder
-          image={secondImage}
-          style={[{ width: secondSize, height: secondSize }, style]}
+          source={secondImage}
+          style={[{ width: secondSize, height: secondSize }, stylesProp?.image]}
           usePlaceholder={usePlaceholder}
         />
       </Animated.View>
+      {children}
     </View>
   )
 }
 
-export default memo(DynamicImage)
+const MemoizedDynamicImage = memo(DynamicImage)
+
+export default MemoizedDynamicImage
```

Comment:
We could just wrap the component itself in `memo` and export that as a named export

Reply:
i think what happens there is that we get an anonymous inner function component, ill experiment and paste some results here

Reply:
So wrapping the component itself: 
![image](https://user-images.githubusercontent.com/8230000/152400744-6f229fb4-5ba1-4cb3-aaa1-d3fc4d3ca27e.png)
And wrapping afterward:
![image](https://user-images.githubusercontent.com/8230000/152401105-4fa1c101-c135-4f53-ac71-0e4fa9ae46cc.png)

Reply:
Ah ok, makes sense. I do think we landed on wrapping the component with memo in one of our product eng syncs, I see us doing it on a few components including Scrubber. We should probably discuss again today so we avoid those Anonymous comps

Reply:
A way to do it inline is this: `const DynamicImage = memo(function DynamicImage({....}) {...}` what do you think about that?

Reply:
I actually kinda like that, we follow a similar pattern in stems with forwardRefs https://github.com/AudiusProject/stems/blob/master/src/components/Button/Button.tsx#L30

Reply:
okay sweet, can def update with a follow up pr here that adds profile page!